cshared.go

```diff
@@ -133,47 +136,79 @@ func FLBPluginInputCallback(data *unsafe.Pointer, csize *C.size_t) int {
 	var err error
 	once.Do(func() {
 		runCtx, runCancel = context.WithCancel(context.Background())
-		theChannel = make(chan Message)
+		// we need to configure this part....
+		theChannel = make(chan Message, 300000)
+		// do we need to buffer this part???
+		cbuf := make(chan Message, 16)
```

Comment:
That is a good question which will require testing, but I believe the buffer is there to avoid deadlocking, especially when there are several coroutines ingesting records.